src/Sylius/Behat/Context/Api/Admin/ManagingChannelsContext.php

```diff
@@ -0,0 +1,249 @@
+    {
+        $this->billingData['taxId'] = $taxId;
+    }
+
+    /**
+     * @When I specify shop billing address as :street, :postcode :city, :country
+     */
+    public function specifyShopBillingAddressAs(
+        string $street,
+        string $postcode,
+        string $city,
+        CountryInterface $country
+    ): void {
+        $this->billingData['street'] = $street;
+        $this->billingData['city'] = $city;
+        $this->billingData['postcode'] = $postcode;
+        $this->billingData['countryCode'] = $country->getCode();
+    }
+
+    /**
+     * @When I add it
+     */
+    public function iAddIt(): void
+    {
+        $this->createBillingData();
+
+        $this->client->create();
+    }
+
+    /**
+     * @When I want to browse channels
+     */
+    public function iWantToBrowseChannels(): void
+    {
+        $this->client->index();
+    }
+
+    /**
+     * @Then I should be notified that it has been successfully created
+     */
+    public function iShouldBeNotifiedThatItHasBeenSuccessfullyCreated(): void
+    {
+        Assert::true(
+            $this->responseChecker->isCreationSuccessful($this->client->getLastResponse()),
+            'Channel could not be created'
+        );
+    }
+
+    /**
+     * @Then the channel :name should appear in the registry
+     * @Then the channel :name should be in the registry
+     */
+    public function theChannelShouldAppearInTheRegistry(string $name): void
+    {
+        Assert::true(
+            $this->responseChecker->hasItemWithValue($this->client->index(),'name', $name),
+            sprintf('Channel with name %s does not exist', $name)
+        );
+    }
+
+    /**
+     * @Then the channel :channel should have :taxon as a menu taxon
+     */
+    public function theChannelShouldHaveAsAMenuTaxon(ChannelInterface $channel, TaxonInterface $taxon): void
+    {
+        Assert::same(
+            $this->responseChecker->getValue($this->client->show($channel->getCode()), 'menuTaxon'),
+            $this->iriConverter->getIriFromItem($taxon),
+            sprintf('Channel %s does not have %s menu taxon', $channel->getName(), $taxon->getName())
+        );
+    }
+
+    /**
+     * @Then I should see :count channels in the list
+     */
+    public function iShouldSeeChannelsInTheList(int $count): void
+    {
+        Assert::same($this->responseChecker->countCollectionItems($this->client->getLastResponse()), $count);
+    }
+
+    private function createBillingData(): void
+    {
+        if (!empty($this->billingData)) {
+            $this->billingDataClient->buildCreateRequest();
+            foreach ($this->billingData as $field => $value) {
+                $this->billingDataClient->addRequestData($field, $value);
+            }
+            $this->billingDataClient->create();
+            $iri = $this->responseChecker->getValue($this->billingDataClient->getLastResponse(), '@id');
+            $this->client->addRequestData('shopBillingData', $iri);
+        }
```

Comment:
I wonder if in our API, there should be a possibility to pass shop billing data config to the channel creation request and do these two calls at once. Wdyt @Sylius/core-team?

If I'm correct this could look like for example adding provinces for countries: https://github.com/Sylius/Sylius/blob/master/src/Sylius/Behat/Context/Api/Admin/ManagingCountriesContext.php#L116:L119